lib/Serialization/DeserializeSIL.cpp

```diff
@@ -667,6 +777,17 @@ SILDeserializer::readSILFunctionChecked(DeclID FID, SILFunction *existingFn,
 
   // If we have an existing function, verify that the types match up.
   if (fn) {
+    assert(!forDebugScope);
+
+    if (fn->isZombie()) {
```

Comment:
Can this branch ever be taken? `fn` shouldn't be a zombie:
* An `existingFn` which is passed to readSILFunctionChecked should either be null or an existing (non-zombie) declaration.
* `SILMod.lookUpFunction(name)` should not return a zombie

Reply:
Aha that makes sense. I think this was taken before I started maintaining the invariant at line 740. I will remove this code and add additional assertions

https://github.com/swiftlang/swift/pull/76934/files#diff-a58e4a75394b751f1446f32a43586ebc1356bf6f94c87affe3e9fb0523287366R740